src/ppx_import.ml

```diff
@@ -182,311 +164,414 @@ let get_modtype_decl ~loc sig_items parent_lid elem =
     | None -> manifest
   in
-  { ptype_name; ptype_params; ptype_kind; ptype_manifest;
-    ptype_cstrs      = [];
-    ptype_private    = Tt.copy_private_flag ttype_decl.type_private;
-    ptype_attributes = Tt.copy_attributes ttype_decl.type_attributes;
-    ptype_loc        = ttype_decl.type_loc; }
+  {
+    ptype_name;
+    ptype_params;
+    ptype_kind;
+    ptype_manifest;
+    ptype_cstrs = [];
+    ptype_private = ttype_decl.type_private;
+    ptype_attributes = ttype_decl.type_attributes;
+    ptype_loc = ttype_decl.type_loc;
+  }
 
-let subst_of_manifest { ptyp_attributes; ptyp_loc ; _ } =
+let subst_of_manifest ({ ptyp_attributes; ptyp_loc; _ } : Ppxlib.core_type) =
+  let open Ppxlib in
   let rec subst_of_expr expr =
     match expr with
-    | [%expr [%e? { pexp_desc = Pexp_ident ({ txt = src ; _ }) ; _ }] :=
-             [%e? { pexp_desc = Pexp_ident (dst); pexp_attributes; pexp_loc; pexp_loc_stack }]] ->
-      [`Lid src, {
-          ptyp_loc = pexp_loc;
-          ptyp_loc_stack = pexp_loc_stack;
-          ptyp_attributes = pexp_attributes;
-          ptyp_desc = Ptyp_constr (dst, []);
-      }]
-    | [%expr [%e? { pexp_desc = Pexp_ident ({ txt = src ; _ }) ; _ }] :=
-             [%e? { pexp_desc = Pexp_ident (dst); pexp_attributes; pexp_loc; pexp_loc_stack; }]; [%e? rest]] ->
-      (`Lid src, {
-         ptyp_loc = pexp_loc;
-         ptyp_loc_stack = pexp_loc_stack;
-         ptyp_attributes = pexp_attributes;
-         ptyp_desc = Ptyp_constr (dst, [])
-      }) :: subst_of_expr rest
-    | { pexp_loc ; _ } ->
-      raise_errorf ~loc:pexp_loc "Invalid [@with] syntax"
+    | [%expr
+        [%e? { pexp_desc = Pexp_ident { txt = src; _ }; _ }]
+        := [%e?
+             {
+               pexp_desc = Pexp_ident dst;
+               pexp_attributes;
+               pexp_loc;
+               pexp_loc_stack;
+             }]] ->
+        [
+          ( `Lid src,
+            {
+              ptyp_loc = pexp_loc;
+              ptyp_loc_stack = pexp_loc_stack;
+              ptyp_attributes = pexp_attributes;
+              ptyp_desc = Ptyp_constr (dst, []);
+            } );
+        ]
+    | [%expr
+        [%e? { pexp_desc = Pexp_ident { txt = src; _ }; _ }]
+        := [%e?
+             {
+               pexp_desc = Pexp_ident dst;
+               pexp_attributes;
+               pexp_loc;
+               pexp_loc_stack;
+             }];
+        [%e? rest]] ->
+        ( `Lid src,
+          {
+            ptyp_loc = pexp_loc;
+            ptyp_loc_stack = pexp_loc_stack;
+            ptyp_attributes = pexp_attributes;
+            ptyp_desc = Ptyp_constr (dst, []);
+          } )
+        :: subst_of_expr rest
+    | { pexp_loc; _ } ->
+        Location.raise_errorf ~loc:pexp_loc "Invalid [@with] syntax"
+  in
+  let find_attr s attrs =
+    try
+      Some
+        (List.find (fun { attr_name = x; _ } -> x.txt = s) attrs).attr_payload
+    with Not_found -> None
```

Comment:
I didn't find the `Ast_convenience.find_attr` equivalent so I just inlined the function